CONTRIBUTING.md

```diff
@@ -123,6 +124,96 @@ from the main (upstream) repository:
     ```
 
 
+## <a name="commit"></a> Commit Message Guidelines
+
+We have very precise rules over how our git commit messages can be formatted.  This leads to **more
+readable messages** that are easy to follow when looking through the **project history**.  But also,
+we use the git commit messages to **generate changelog**.
+
+### Commit Message Format
+Each commit message consists of a **header**, a **body** and a **footer**.  The header has a special
+format that includes a **type**, a **scope** and a **subject**:
+
+```
+<type>(<scope>): <subject>
+<BLANK LINE>
+<body>
+<BLANK LINE>
+<footer>
+```
+
+The **header** is mandatory and the **scope** of the header is optional.
+
+Any line of the commit message cannot be longer 100 characters! This allows the message to be easier
+to read on GitHub as well as in various git tools.
+
+### Revert
+If the commit reverts a previous commit, it should begin with `revert: `, followed by the header of
+the reverted commit. In the body it should say: `This reverts commit <hash>.`, where the hash is
+the SHA of the commit being reverted.
+
+### Type
+Must be one of the following:
+
+* **feat**: A new feature
+* **fix**: A bug fix
+* **docs**: Documentation only changes
+* **style**: Changes that do not affect the meaning of the code (white-space, formatting, missing
+  semi-colons, etc)
+* **refactor**: A code change that neither fixes a bug nor adds a feature
+* **perf**: A code change that improves performance
+* **test**: Adding missing tests or correcting existing tests
+* **build**: Changes that affect the build system, CI configuration or external dependencies
+            (example scopes: gulp, broccoli, npm)
+* **chore**: Other changes that don't modify `src` or `test` files
+
+### Scope
```

Comment:
I think if we want to use scopes we should determine their names - otherwise there will be very common for one commit to belong to many scopes.
But If we decide to put only some commit messages into changelog scopes might become unnecessary at all.

Reply:
Maybe define just `core` and all drivers as scopes for now? Or just omit this part of commit message.

Reply:
I think we could omit it for now. It will be useful when we split codebase into multiple packages(if we decide to go with monorepo approach).